rgo/component.py

```diff
@@ -30,7 +30,7 @@
 SRPM_RE = re.compile(r".+\.(?:no)?src\.rpm")
 
 class Component(object):
-    def __init__(self, name, version_from=None, git=None, distgit=None, requires=[]):
+    def __init__(self, name, version_from=None, git=None, distgit=None, requires=[], overrides=[]):
```

Comment:
Using mutable default arguments (`overrides=[]`) can have nasty side effects. We should probably avoid it. See https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/ for details.

Reply:
Oh, this. Yeah, if you don't mutate the value of the argument, its fine :) In this case it's "only" copied to a `set`.

But it might be better to avoid this altogether...

Reply:
> If that's possible with the way you inherit the `Overrides` from `DistGit`.
> 
> Alternatively, call it `distgit-overrides` if we'd keep it under `component` in the yaml?

It looks like its not possible so it will have to be `distgit-overrides`.

Just throwing an idea but I was thinking another option could be dropping the overrides concept and instead doing:
```
components:
  - name: foo
    git:
      src: github:someone/foo.git
    distgits:
      - src: fedorapkgs:foo.git
        branch: origin/f32
        patches: drop
      - src: centos-distgit:foo.git
        branch: c8s
        spec-path: SPECS/foo.spec
        roots:
          - centos-stream-8-x86_64
```
It is not backwards compatible (if that matters?) and I am not sure if no `roots` should mean all default builds.
Though it is simpler I think.

Reply:
I think `distgit-overrides` is more explicit, your suggestion would need the rule that if there are no `roots` it's the default, and then what if there are more items in the array that have no `roots`...